packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -86,4 +140,137 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
             order_id: orderID,
         });
     }
+
+    private async _onHostedMethodApprove(data: ApproveDataOptions, actions: ApproveActions) {
+        try {
+            const orderDetails = await actions.order.get();
+            const consignments = this._store.getState().consignments.getConsignmentsOrThrow();
+            const lineItems = this._getLineItems();
+            if (!this._paymentMethod?.id) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+            const methodId = this._paymentMethod?.id;
+            const shippingAddress = {
+                ...this._currentShippingAddress,
+                firstName: orderDetails.payer.name.given_name,
+                lastName: orderDetails.payer.name.surname,
+                email: orderDetails.payer.email_address,
+                address1: orderDetails.purchase_units[0].shipping.address.address_line_1,
+                address2: '',
+                stateOrProvince: '',
+                stateOrProvinceCode: '',
+                city: this._currentShippingAddress?.city || '',
+                countryCode: this._currentShippingAddress?.countryCode || '',
+                postalCode: this._currentShippingAddress?.postalCode || '',
+                phone: '',
+                company: '',
+                customFields: []
+            };
```

Comment:
As far as I see, we can make this code a bit simpler. 
`this._currentShippingAddress` - is a result of `_transformAddress` method. We are mocking almost all the data in `_transformAddress` method and we can skip mock duplication here.

In this case we can get next result:
 ```suggestion
            const shippingAddress = {
                ...this._currentShippingAddress,
                firstName: orderDetails.payer.name.given_name,
                lastName: orderDetails.payer.name.surname,
                email: orderDetails.payer.email_address,
                address1: orderDetails.purchase_units[0].shipping.address.address_line_1,
            };
```